### DISCUSSION TITLE
Confusion and clarification for reference types

### DISCUSSION THREAD

**Discussion by Rycko1** [supabase-community/supabase-csharp#162] [Q&A]
Hi! 

I have a deeply interconnected schema with a lot of foreign keys on various tables. I am trying to handle these using the [Reference] attribute but I am running into some problems, likely due to me simply not understanding how the attribute works correctly.

Given the following models
```
[Table("foo")]
public class Foo(Bar bar): BaseModel
{
    [PrimaryKey("id")] public Guid Id { get; set; } = Guid.NewGuid();

    [Reference(typeof(Bar))]
    public Bar Bar { get; set; } = bar;
}

[Table("bar")]
public class Bar: BaseModel
{
    [PrimaryKey("id")] public Guid Id { get; set; } = Guid.NewGuid();

    [Column("name")]
    public string name { get; set; } = "test123;
}

```

And the database schema
```
create table public.bar
(
 id uuid primary key default uuid_generate_v4(),
 name text
)

create table public.foo
(
 id uuid primary key default uuid_generate_v4(),
 bar_id uuid references bar(id) not null
)
```

How can I insert into the table `foo` and set the foreign key of `bar`? I have tried a few different things but nothing seems to work correctly

I have tried
- Adding `[Column]` attribute to the reference, this results in the error `invalid input syntax for type uuid` from postgrest
- Adding a separate `public Guid bar_id` property to the `foo` model and populating this separately for inserts, this works somewhat fine but is messy, requiring two unique properties on every model, one for queries and one for inserts, there has to be a better way

Are there any examples out there of inserting foreign key data with the models and schemas?

**Comment by luke-ransfield** [comment]
I am far from an expert but I think this discussion might be what you're looking for:

https://github.com/supabase-community/supabase-csharp/discussions/123
